spyder/preferences/shortcuts.py

```diff
@@ -122,47 +156,77 @@ def keyPressEvent(self, event):
 
 class ShortcutEditor(QDialog):
     """A dialog for entering key sequences."""
+
     def __init__(self, parent, context, name, sequence, shortcuts):
         super(ShortcutEditor, self).__init__(parent)
         self._parent = parent
 
         self.context = context
-        self.npressed = 0
-        self.keys = set()
-        self.key_modifiers = set()
-        self.key_non_modifiers = list()
-        self.key_text = list()
-        self.sequence = sequence
-        self.new_sequence = None
-        self.edit_state = True
+        self.name = name
         self.shortcuts = shortcuts
+        self.cur_sequence = sequence
 
+        self.new_subseq = list()
+        self.new_sequence = ', '.join(self.new_subseq)
+        self.new_qsequence = QKeySequence(self.new_sequence)
+
+        self.setup()
+
+    def setup(self):
+        """Setup the ShortcutEditor with the provided arguments."""
         # Widgets
         self.label_info = QLabel()
-        self.label_info.setText(_("Press the new shortcut and select 'Ok': \n"
-             "(Press 'Tab' once to switch focus between the shortcut entry \n"
-             "and the buttons below it)"))
+        self.label_info.setText(
+            _("Press the new shortcut and select 'Ok' "
+              "or press 'Clear' to unbind the shortcut."))

```

Comment:
Press the new shortcut and select ``Ok`` to confirm,
click ``Cancel`` to revert to the previous state,
or use ``Clear`` to unbind the command from a shortcut.